### ISSUE TITLE
Upgrading docker then unable to run old images due to repository name restrictions

### ISSUE THREAD

**Issue by iskradelta** [docker/docker#16393]
Hi,

See we had an image named like this awesome.com/project/abc_12def3_4_5_6_-_gh-cuont-lol

Old docker client was at 1.6 (even used when at 1.3), suddenly upgraded to 1.8, because why not right, its not a major version change? Right guys?

And now:
repository name component must match "[a-z0-9]+(?:[._-][a-z0-9]+)*"

Its the _-_ which doesnt match that regexp. Which sucks, because why should all our old images suddenly be considered invalid?

**Comment by iskradelta** [comment]
$ docker version
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Sat Aug 15 17:29:10 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Sat Aug 15 17:29:10 UTC 2015
 OS/Arch:      linux/amd64



$ docker info
Containers: 17
Images: 82
Storage Driver: devicemapper
 Pool Name: docker-8:1-1975770-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop2
 Metadata file: /dev/loop3
 Data Space Used: 11.4 GB
 Data Space Total: 107.4 GB
 Data Space Available: 5.492 GB
 Metadata Space Used: 10.69 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.137 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.102 (2015-07-07)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.7.201509131604-1-grsec
Operating System: Arch Linux
CPUs: 8
Total Memory: 27.39 GiB
Name: yourmother
ID: OZT6:HVEQ:CIK6:HHQX:OF3V:EKVA:HUQT:3PYJ:2MEV:FIJS:PVLI:CHAH


$ uname -a
Linux precizija 4.1.7.201509131604-1-grsec #1 SMP PREEMPT Mon Sep 14 00:45:54 EDT 2015 x86_64 GNU/Linux

**Comment by cpuguy83** [comment]
ping @tiborvass 
Looks like this is coming from the v2 API client from distribution?

**Comment by stevvooe** [comment]
`awesome.com/project/abc_12def3_4_5_6_-_gh-cuont-lol` is simply not a valid image name. Such image names were only valid due to weak validation. Older docker daemons were inconsistent in their validation of image names and we've taken the task of clearly defining this.

Can you not simply rename these to something else?

**Comment by iskradelta** [comment]
@stevvooe can you explain the 'simply' part? how is it simply not a valid image name?

**Comment by dmcgowan** [comment]
@iskradelta the repository name regex represents the image format that will supported  going forward. If you are having issues migrating existing images we can advise how to migrate such names. We can also have the discussion about expanding the regexp as well by opening a PR against distribution. 

Why? We are moving to a more clearly defined format to avoid these situations in the future. Right now it is hard to parse and match image names without such a format.

**Comment by icecrime** [comment]
Closed by #16867.

**Comment by josephbergevin** [comment]
I'm having the same issue with this image name: nz--urls
Any chance you could allow 2 dashes in a row in the image name regex?

**Comment by stevvooe** [comment]
@josegonzalez Please avoid commenting on closed issues. This has been fixed by #16867 and should be out in the next release.

**Comment by josegonzalez** [comment]
@stevvooe I'm not even sure what I commented since it was deleted, but okay, I'll keep that in mind going forward.

**Comment by stevvooe** [comment]
@josegonzalez I think you asked about using some "n--x" name.

**Comment by stevvooe** [comment]
@josegonzalez Even worse. I think it was @josephbergevin. Sorry about that.

I'll discontinue violating the policy of commenting on closed issues. ;)

**Comment by josephbergevin** [comment]
Yep. Sorry about that. I'll never do it again :)

**Comment by stevvooe** [comment]
@josephbergevin No worries.

This is the last time, I swear.

:dog:
